src/modules/filter-drawer/filter-drawer.tsx

```diff
@@ -0,0 +1,349 @@
+        duration: 300,
+      }),
+
+      // Header Back
+      Animated.timing(headerBackTranslateX, {
+        toValue: -windowWidth,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+      Animated.timing(headerBackOpacity, {
+        toValue: 0,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+    ]).start();
+  };
+
+  // On Show Secondary Content
+  const onShowSecondaryContent = (): void => {
+    Animated.parallel([
+      // Header Back
+      Animated.timing(headerBackOpacity, {
+        toValue: 1,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+      Animated.timing(headerBackTranslateX, {
+        toValue: 0,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+
+      // Header Content
+      Animated.timing(headerContentOpacity, {
+        toValue: 1,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+      Animated.timing(headerContentTranslateX, {
+        toValue: -windowWidth,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+
+      // Close Button
+      Animated.timing(closeButtonOpacity, {
+        toValue: 0,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+
+      // Content Wrapper
+      Animated.timing(contentWrapperTranslateX, {
+        toValue: -windowWidth,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+    ]).start();
+  };
+
+  // On Close Modal
+  const onCloseModal = () => {
+    // @ts-ignore
+    modalRef?.current?.close?.();
+
+    // Call callback function
+    onClose();
+  };
+
+  React.useEffect(() => {
+    if (shouldShowSecondaryContent) {
+      onShowSecondaryContent();
+    } else {
+      onShowPrimaryContent();
+    }
+  }, [shouldShowSecondaryContent]);
+
+  const AnimatedWrapper = Animated.createAnimatedComponent(ContentWrapper);
+  const AnimatedCloseWrapper =
+    Animated.createAnimatedComponent(CloseButtonWrapper);
+
+  const modalStyle = {margin: 0};
+
+  return (
+    <Modal
+      ref={modalRef}
+      isVisible={isOpen}
+      onModalHide={onClose}
+      hideModalContentWhileAnimating
+      useNativeDriverForBackdrop
+      style={modalStyle}
+      useNativeDriver
+      propagateSwipe>
+      <Wrapper testID="drawer-wrapper">
+        {/* Sections */}
+        <SectionWrapper>
+          {/* Content */}
+          <AnimatedWrapper
+            style={{transform: [{translateX: contentWrapperTranslateX}]}}>
+            {/* Primary Content */}
+            <PrimaryContentWrapper testID="drawer-primary-content">
+              <ScrollView showsVerticalScrollIndicator={false}>
+                {/* Header */}
+                <Header.Wrapper>
+                  <Animated.View
+                    style={{
+                      opacity: headerContentOpacity,
+                      transform: [{translateX: headerContentTranslateX}],
+                    }}>
+                    <Header.Title variant="h2">Filters</Header.Title>
```

Comment:
Please make sure that filters title should have top margin of 33px. and can you also make this title prop dynamic?